pipeline.go

```diff
@@ -50,26 +50,38 @@ func (c *Pipeline) Discard() error {
 
 // Exec always returns list of commands and error of the first failed
 // command if any.
-func (c *Pipeline) Exec() ([]Cmder, error) {
+func (c *Pipeline) Exec() (cmds []Cmder, retErr error) {
```

Comment:
My worry is that `isNetworkError` only focusses on the kind of error but not if it occurred during request or response. There is a chance of a network disconnect while reading a response, which could still mean that the request succeeded. If you have e.g. 1M `INCR` commands in your pipeline, retrying suddenly becomes a dangerous business.